### ISSUE TITLE
Adding an option in Generate Code to include defaults arguments as imported from special.json

### ISSUE THREAD

**Issue by Demae** [DottieDot/GTAV-NativeDB#54]
^

**Comment by DottieDot** [comment]
I'm actually planning on fully intergrating the special.json eventually but I've been putting it off for a while.

**Comment by DottieDot** [comment]
Some challenges with implementing this:
* Not everyone's `special.json` is the same.
  * Can an unknown type determined to be an `ENUM` or a `STRUCT`?
* Some default values consist of expressions:
  * Constants.
  * Enum values.
  * Logical/arithmetic expressions on constants.
* For C++ the types and constsants have to be declared in the correct order.
* Should this work with "use native types"? 
* Enums only assume the vanilla game, how should extra content be modeled for enums (e.g. model names)?
  * For TypeScript: `ENUM_NAME | number`
  * For Rust: `num_enum`'s `catch_all`
  * For C++: ?
  * For C# (If relevant): ?

Actions to be completed before this can be implemented:
* [ ] Extend https://github.com/DottieDot/gta-5-native-documentation-generator:
  * Remove unused declarations (but what about constants that should be used with natives?).
  * Sort for declaration order.
  * Model expressions.

**Comment by Demae** [comment]
Yes, speaking of the special.json generator, it worked well for most of the .sch files however when it hits an ifdef such as #IF IS_DEBUG_BUILD it breaks and skips to the next .sch

**Comment by Demae** [comment]
I'm not familiar with TS or Rust, but with the json schema implemented it should be relatively easy to control the input, and as far as extra content goes, I've found it easier to include everything regardless of game version.

**Comment by DottieDot** [comment]
> I'm not familiar with TS or Rust, but with the json schema implemented it should be relatively easy to control the input, and as far as extra content goes, I've found it easier to include everything regardless of game version.

The problem is that this will break some enums and structs. For example, some enums will have something like this:
```
ENUM FOO
  // Fields...
#IF FLAG
  SOMETHING, 
#ENDIF
#IF NOT FLAG
  UNUSED,
#ENDIF
  // More fields...
ENDENUM
```

Which is why this problem exists:
> Yes, speaking of the special.json generator, it worked well for most of the .sch files however when it hits an ifdef such as #IF IS_DEBUG_BUILD it breaks and skips to the next .sch

Because making a proper for system for this is a good amount of effort.

**Comment by Demae** [comment]
and ignoring them would be a bandaid, I see, well considering that generating the json is on a user-by-user basis, would it be an idea to add an option to the generator for users to define which #IFs they want defined and the site handles it as it is?

**Comment by DottieDot** [comment]
> would it be an idea to add an option to the generator for users to choose which #IFs they want defined and the site handles it as it is?

That's the thing that would be a good amount of effort, and since most people will only do this once it doesn't seem worth it to me. However, I'm considering making some other changes at which point I could possibly add it.
